src/status.rs

```diff
@@ -0,0 +1,95 @@
+//! A Status Report announces the station's current mission or any other single
+//! line status to everyone. The report starts with the '>' APRS Data Type Identifier.
+//! The report may optionally contain a timestamp.
+//!
+//! Examples:
+//! - ">12.6V 0.2A 22degC"              (report without timestamp)
+//! - ">120503hFatal error"             (report with timestamp in HMS format)
+//! - ">281205zSystem will shutdown"    (report with timestamp in DHM format)
+
+use std::convert::TryFrom;
+use std::io::Write;
+
+use AprsError;
+use EncodeError;
+use Timestamp;
+
+#[derive(PartialEq, Debug, Clone)]
```

Comment:
Is there a reason to derive `PartialEq` but not `Eq`?

Reply:
No... I just took this line from `AprsPosition` ... I see, that AprsMessage derives from Eq _and_ PartialEq. What is the right way for these structs?

Reply:
[`Eq`](https://doc.rust-lang.org/std/cmp/trait.Eq.html) is a superset of [`PartialEq`](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html) which promises that the comparison obeys the commutative, reflexive, and transitive properties. In other words, `Eq` is an [equivalence relation](https://en.wikipedia.org/wiki/Equivalence_relation), while `PartialEq` is just a [partial equivalence relation](https://en.wikipedia.org/wiki/Partial_equivalence_relation).

An example of why the distinction matters is `HashMap`, which requires that it's keys implement `Eq`. If it didn't have this requirement, you could add keys that didn't obey the reflexive property (`x==x`), so the hash map wouldn't have a reliable way to tell if two keys are actually equivalent.

The typical example of something that implements `PartialEq` but not `Eq` is floating point numbers. `NaN != NaN`, which violates the reflexive property. This why it's not implemented for `AprsPosition`, since that struct contains floats.

In this case, the `cmp` implementation from `#[derive(PartialEq)]` is a full equivalence relation, so we should implement `Eq`. In general, it's always safe to try `#[derive(Eq, PartialEq)]`. If you have a field has a type that doesn't implement `Eq` (like `f32`) you'll just get a compile error rather than accidentally implementing `Eq` for something that doesn't really have a full equivalence relation. If your type has generic parameters, the derive implementation will generate the appropriate bounds as well.

Reply:
Thank you very much for the explanation. Just fixed it